### PR TITLE
Report style errors at the correct location.

### PR DESCRIPTION
**Pull request by davidcian**
A space has been removed from an if statement in StandardCompiler.cpp, at line 227, thus introducing a coding style violation. The purpose is to see how CircleCI reacts to this. Next, I will try to get CircleCI to comment about this line directly on the diff.

I do not mean for this to be merged, I just couldn't think of any other way to test CircleCI's behavior.

### CODE REVIEW

**Comment by chriseth** [comment]
Hey David! Do you need any help here?

**Comment by davidcian** [comment]
> Hey David! Do you need any help here?

Hey! Actually if possible a bit, yes. I've managed to post the diff review, but I'm off by about 10 lines usually (by just using the line number from the check style script), and I'm not sure why or where to look for line number rules in GitHub.

For reference, here is a PR on my fork with this behavior: https://github.com/davidcian/solidity/pull/1

**Comment by davidcian** [comment]
@chriseth I think everything should be alright now, although I suspect I've missed a couple of details. The bot posts diff review comments with the text "Coding style error" on the affected lines, one comment per violation.

File: scripts/report_errors.sh [hunk 1]
```diff
@@ -44,6 +43,26 @@ function post_error_to_github
       --header 'content-type: application/json' \
       -u stackenbotten:$GITHUB_ACCESS_TOKEN \
       --data "{\"body\": \"There was an error when running \`$CIRCLE_JOB\` for commit \`$CIRCLE_SHA1\`:\n\`\`\`\n$FORMATTED_ERROR_MSG\n\`\`\`\nPlease check that your changes are working as intended.\"}"
+
+    post_review_comment_to_github
+}
+
+function post_review_comment_to_github {
```
Comment (chriseth) [inline scripts/report_errors.sh]:
```suggestion
function post_review_comment_to_github
{
```

File: scripts/report_errors.sh [hunk 2]
```diff
@@ -44,6 +43,26 @@ function post_error_to_github
...
+    GITHUB_API_URL="https://api.github.com/repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/pulls/$CIRCLE_PR_NUMBER/comments"
+
+    sed -i 1d $ERROR_LOG
```
Comment (chriseth) [inline scripts/report_errors.sh:55]: What does this do?

File: scripts/report_errors.sh [hunk 3]
```diff
@@ -44,6 +43,26 @@ function post_error_to_github
...
+
+    while read line; do
```
Comment (chriseth) [inline scripts/report_errors.sh]:
```suggestion
    while read line
    do
```

File: scripts/report_errors.sh [hunk 4]
```diff
@@ -44,6 +43,26 @@ function post_error_to_github
...
+        TESTING_PATH=$(echo $line | grep -oE ".*\.cpp")
```
Comment (chriseth) [inline scripts/report_errors.sh]: Rather name this `ERROR_PATH` and `ERROR_LINE`?

**Review by chriseth — CHANGES_REQUESTED** [review]
Looks good, thanks a lot! Can you please squash into a single commit, remove `.circleci_config.orig` and do some of the style changes? Thanks!

File: scripts/report_errors.sh (hunk 2 above)
Comment (davidcian) [reply scripts/report_errors.sh:55]: This uses the stream edit (sed) command to remove the first line from the error log in place (without creating a new file). It is done to prepare for further processing using grep, as the first line just announces the fact that there are errors. The next lines in the error log contains errors, one per line.

File: scripts/report_errors.sh (hunk 4 above)
Comment (davidcian) [reply scripts/report_errors.sh]: Ah yes, leftover from earlier changes, good point

**Comment by chriseth** [comment]
Can you squash to a single commit, please?

**Comment by davidcian** [comment]
> Can you squash to a single commit, please?

I've done my best, but for some reason I'm not sure how to further squash the remaining commits. When I do git log or even when I attempt to git rebase interactive with the commit hash of the first commit, I can't find the three remaining commits in the opened editor so I can squash them.

**Comment by leonardoalt** [comment]
The problem probably is that `develop` was merged into the branch and not rebased.

**Review by chriseth — APPROVED** [review]

**Review by chriseth — APPROVED** [review]

File: scripts/report_errors.sh [hunk 5]
```diff
@@ -44,6 +44,28 @@ function post_error_to_github
...
+
+function post_review_comment_to_github
+{
+    GITHUB_API_URL="https://api.github.com/repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/pulls/$CIRCLE_PR_NUMBER/comments"
+
+    sed -i 1d $ERROR_LOG
+
+    while read line
+    do
+        ERROR_PATH=$(echo $line | grep -oE ".*\.cpp")
```
Comment (chriseth) [inline scripts/report_errors.sh:59]: I think the regexes here can still be improved, but I'm merging for now. Thanks a lot for your help!

**Review by chriseth — APPROVED** [review]

**Pull request merged**